src/Microsoft.Kiota.Cli.Commons/Extensions/CommandBuilderExtensions.cs

```diff
@@ -69,4 +70,77 @@ public static CommandLineBuilder RegisterCommonServices(this CommandLineBuilder
         });
         return builder;
     }
+
+    /// <summary>
+    /// Registers a headers option to all executable commands in the builder
+    /// and a middleware that reads options from the parsed result and adds
+    /// them to the store instance returned after calling
+    /// <paramref name="headersStoreGetter"/>.
+    /// </summary>
+    /// 
+    /// <param name="builder">A command line builder</param>
+    /// <param name="headersStoreGetter">
+    /// A delegate that will be called to get an instance of
+    /// <see cref="IHeadersStore"/>.
+    /// </param>
+    /// <param name="name">
+    /// A custom name for the registered option.
+    /// Defaults to <c>--headers</c>
+    /// </param>
+    /// <param name="customDescription">
+    /// An optional description for the registered headers option.
+    /// </param>
+    /// 
+    /// <returns>
+    /// The same instance of <see cref="CommandLineBuilder"/>.
+    /// </returns>
+    /// 
+    /// <remarks>
+    /// <para>
+    /// This function must be called after the root command has been set in
+    /// the <see cref="CommandLineBuilder"/>.
+    /// </para>
+    /// <para>
+    /// This function does nothing to add the headers to a request. You would
+    /// need to implement a way to read from the updated
+    /// <see cref="IHeadersStore"/> and update the request. For an example that
+    /// works with <see cref="System.Net.Http.HttpClient"/>,
+    /// see the
+    /// <see cref="Microsoft.Kiota.Cli.Commons.Http.NativeHttpHeadersHandler"/>
+    /// class.
+    /// </para>
+    /// </remarks>
+    public static CommandLineBuilder RegisterHeadersOption(this CommandLineBuilder builder, Func<IHeadersStore> headersStoreGetter, string name="--headers", string? customDescription = null)
+    {
+        // TODO: Check for empty name?
```

Comment:
should clean up

Reply:
I'll clean up before merge. I'm researching whether these checks are done by the system.commandline library already.